2-dom-manipulation/form.js

```diff
@@ -0,0 +1,71 @@
+const form = document.querySelector("form");
+const nameField = document.querySelector("#name");
+const lastNameField = document.querySelector("#lastName");
+const emailField = document.querySelector("#email");
+const phoneNumberField = document.querySelector("#phone");
+const radioInputs = document.querySelectorAll("#estrato");
+const inputs = document.querySelectorAll("input");
+
+const expressions = {
+  name: /^([a-zA-Z]+\s?){2}?$/,
+  email: /^[a-zA-Z0-9._-]+@[a-zA-z0-9._-]+\.[a-zA-Z]+$/,
+  phoneNumber:
+    /^(\+\d{2})?[\s\-]?(\d{3})[\s\-]?(\d{2,3})[\s\-]?(\d{2})[\s\-]?(\d{2})$/,
+};
+
+function validateForm(event) {
+  switch (event.target.name) {
+    case "name":
+      if (expressions.name.test(event.target.value)) {
+        nameField.classList.add("correctField");
+        nameField.classList.remove("errorField");
+      } else {
+        nameField.classList.add("errorField");
+        nameField.classList.remove("correctField");
+      }
+      break;
+    case "lastName":
+      if (expressions.name.test(event.target.value)) {
+        lastNameField.classList.add("correctField");
+        lastNameField.classList.remove("errorField");
+      } else {
+        lastNameField.classList.add("errorField");
+        lastNameField.classList.remove("correctField");
+      }
+      break;
+    case "email":
+      if (expressions.email.test(event.target.value)) {
+        emailField.classList.add("correctField");
+        emailField.classList.remove("errorField");
+      } else {
+        emailField.classList.add("errorField");
+        emailField.classList.remove("correctField");
+      }
+      break;
+    case "phone":
+      if (expressions.phoneNumber.test(event.target.value)) {
+        phoneNumberField.classList.add("correctField");
+        phoneNumberField.classList.remove("errorField");
+      } else {
+        phoneNumberField.classList.add("errorField");
+        phoneNumberField.classList.remove("correctField");
+      }
+      break;
+  }
+}
+
+inputs.forEach((input) => {
+  input.addEventListener("keyup", validateForm);
+  input.addEventListener("blur", validateForm);
+});
```

Comment:
The problem with this approach is that if you load the page and you hit the submit button, you will be able to skip all the validations. Make sure the validations are completed when the form is submitted as well.